sdks/python/apache_beam/yaml/yaml_provider.py

```diff
@@ -59,7 +60,11 @@ def provided_transforms(self) -> Iterable[str]:
     raise NotImplementedError(type(self))
 
   def create_transform(
-      self, typ: str, args: Mapping[str, Any]) -> beam.PTransform:
+      self,
+      typ: str,
+      args: Mapping[str, Any],
+      yaml_create_transform: Callable[[Mapping[str, Any]], beam.PTransform]
```

Comment:
FYI, I'm looking at refactoring this a bit (doing a re-write like how the other "meta" transforms are handled, as this is a bit hacky.

Reply:
This turns out to have its own non-trivial complications. Keeping as is.